graphql/schemabuilder/pagination.go

```diff
@@ -341,6 +400,42 @@ func (sb *schemaBuilder) getKeyFieldOnStruct(nodeType reflect.Type) (string, err
 
 }
 
+// Parses the return types and checks if there's a pageInfo struct being returned by the resolver
+func (funcCtx *funcContext) parsePaginatedReturnSignature(m *method) (retConnInfo bool, err error) {
+	retConnInfo = false
+
+	out := make([]reflect.Type, 0, funcCtx.funcType.NumOut())
+	for i := 0; i < funcCtx.funcType.NumOut(); i++ {
+		out = append(out, funcCtx.funcType.Out(i))
+	}
+
+	if len(out) > 0 && out[0] != errType {
+		funcCtx.hasRet = true
+		out = out[1:]
+	}
+
+	if len(out) > 0 && out[0] == reflect.TypeOf(PaginationInfo{}) {
+		retConnInfo = true
```

Comment:
should we error if the resolver takes pagination args but doesn't return PaginationInfo?

Reply:
also nit: can we keep this variable name consistent between `ConnInfo` and `PaginationInfo`?